pandas/core/series.py

```diff
@@ -485,6 +471,16 @@ def dtypes(self):
         """
         return self._data.dtype
 
+    @property
+    def name(self) -> Hashable:
```

Comment:
@WillAyd @simonjayhawkins do you have a preferred type on Series.name? Previously I had `Optional[Hashable]`, but None is Hashable so either works.

In DataFrameGroupBy._iterate_slices we have its type as just `Hashable`.

Reply:
I think this is fine but no strong preference

Reply:
just ran into problems with this in `Series._unpickle_series_compat`, 

`pandas\core\series.py:1018:25: error: Incompatible types in assignment (expression has type "Optional[Any]", variable has type "Hashable")  [assignment]`

so will need to change. see https://github.com/pandas-dev/pandas/pull/28339/commits/991e90d50d4c0a4a3ad4ff699e208a263dd76515

```python
from typing import Optional, Hashable

name: Optional[Hashable]
name2: Hashable

name = None
name2 = None  # error: Incompatible types in assignment (expression has type "None", variable has type "Hashable")  [assignment]
```